File: Formula1/src/Formula1/JAVA/Monoplaza.java
package Formula1.JAVA;

public abstract class Monoplaza {
    protected String nombrePiloto;
    protected String nacionalidad;
    protected int numeroPlaza;
    protected String escuderia;
    protected int cantidadPuntos;


    ////////////CONSTRUCTOR/////////////////
            ///CON PARAMETROS///
    public Monoplaza(String pNombrePiloto, String pNacionalidad, int pNumeroPlaza,String pEscuderia ,int pCantidadPuntos){
    this.nombrePiloto=pNombrePiloto;
    this.nacionalidad=pNacionalidad;
    this.numeroPlaza=pNumeroPlaza;
    this.escuderia=pEscuderia;
    this.cantidadPuntos=pCantidadPuntos;
    }

    ///////////////////GETTERS Y SETTERS/////////////////
    public String getNombrePiloto() {
        return nombrePiloto;
    }
    public void setNombrePiloto(String nombrePiloto) {
        this.nombrePiloto = nombrePiloto;
    }
    public String getNacionalidad() {
        return nacionalidad;
    }
    public void setNacionalidad(String nacionalidad) {
        this.nacionalidad = nacionalidad;
    }
    public int getNumeroPlaza() {
        return numeroPlaza;
    }
    public void setNumeroPlaza(int numeroPlaza) {
        this.numeroPlaza = numeroPlaza;
    }
    public String getEscuderia() {
        return escuderia;
    }
    public void setEscuderia(String escuderia) {
        this.escuderia = escuderia;
    }
    public int getCantidadPuntos() {
        return cantidadPuntos;
    }
    public void setCantidadPuntos(int cantidadPuntos) {
        this.cantidadPuntos = cantidadPuntos;
    }

    /////////////////METODOS////////////////////
    
    public abstract void otorgarPuntos(int posicion, boolean vueltaRapida);
    public abstract boolean posicionValida(int posicion);

    @Override
    public String toString() {
        return "Piloto: " + nombrePiloto + " (" + nacionalidad + "), Nº " + numeroPlaza +
                ", Escudería: " + escuderia + ", Puntos: " + cantidadPuntos;
    }
    
}

    

File: Formula1/src/Formula1/PHP/Monoplaza.php
<?php
namespace Formula1;

abstract class Monoplaza {
    protected  $nombrePiloto;
    protected  $nacionalidad;
    protected  $numeroPlaza;
    protected  $escuderia;
    protected  $cantidadPuntos;
    
    ////CONSTRUCTOR///
    public function __construct(string $pNombrePiloto, string $pNacionalidad, int $pNumeroPlaza, string $pEscuderia, int $pCantidadPuntos) {
        $this->nombrePiloto = $pNombrePiloto;
        $this->nacionalidad = $pNacionalidad;
        $this->numeroPlaza = $pNumeroPlaza;
        $this->escuderia = $pEscuderia;
        $this->cantidadPuntos = $pCantidadPuntos;
    }

    // GETTERS Y SETTERS ////
    public function getNombrePiloto(){ 
        return $this->nombrePiloto; 
    }
    public function setNombrePiloto($pNombrePiloto){ 
        $this->nombrePiloto = $pNombrePiloto; 
    }

    public function getNacionalidad(){ 
        return $this->nacionalidad; 
    }
    public function setNacionalidad($pNacionalidad){
        $this->nacionalidad = $pNacionalidad; 
    }

    public function getNumeroPlaza(){ 
        return $this->numeroPlaza; 
    }
    public function setNumeroPlaza($pNumeroPlaza){
        $this->numeroPlaza = $pNumeroPlaza; 
    }

    public function getEscuderia(){
        return $this->escuderia; 
    }
    public function setEscuderia($pEscuderia){
        $this->escuderia = $pEscuderia;
    }

    public function getCantidadPuntos(){ 
        return $this->cantidadPuntos; 
    }
    public function setCantidadPuntos($pCantidadPuntos){
        $this->cantidadPuntos = $pCantidadPuntos; 
    }

    // METODOS ///
    abstract public function otorgarPuntos($posicion,$vueltaRapida);
    abstract public function posicionValida($posicion);

    ///TOSTRING///
    public function __toString(){
    return "Piloto: {$this->nombrePiloto}" .
        ", Nacionalidad: {$this->nacionalidad}" .
        ", Nº: {$this->numeroPlaza}" .
        ", Escudería: {$this->escuderia}" .
        ", Puntos: {$this->cantidadPuntos}";
    }
}